### DISCUSSION TITLE
Problem with BVAnalyzer and calculate_bv_sum

### DISCUSSION THREAD

**Discussion by smarwin** [materialsproject/pymatgen#3103] [Q&A]
Hey all,

I am trying to calculate BVS values for a big set of cif files. For single files I have used EXPO2014 and that is also what I use for reference if the BVS values from pymatgen "make sense". Unfortunately, the values I get from calculate_bv_sum are very much off of what to expect. On the example of P5+ I get a mean value of 6 and for Si4+ a mean value of 5. From time to time I also get Ba2+ values of 4. That does not happen with EXPO2014.

Now, I am wondering if that is just a "scaling problem"? There is a scale_factor parameter that defaults to 1. Is it okay to adjust this value in a way that I get "good values"? Or any other suggestions?

Here is a working (minimal) code snippet of what I use:
```
# Imports
from pymatgen.core import Structure
from pymatgen.analysis.bond_valence import calculate_bv_sum

def bvs_calculator(structure):
    # Loop through all sites in structure
    for site in structure.sites:
        # Get neighbors of site
        nn_list = structure.get_neighbors(site, 4)
        # Calculate bond valence sum of site
        bvs_value = calculate_bv_sum(site, nn_list)
        # Print site specie and bond valence sum
        print(site.specie, bvs_value)

file_path = 'some_cif_cile.cif'
structure = Structure.from_file(file_path)
bvs_calculator(structure)
```

I'd appreciate your help big time!
Cheers,
Marwin

**Comment by mkhorton** [comment]
Hi Marwin, the BVAnalyzer code is actually one of the few corners of pymatgen I don't have a good understanding of, so I'd like to understand this better.

I notice the BV dataset is actually quite old (doi: 10.1021/ja00009a002), and that newer datasets are available -- could this be part of the cause?

I vaguely remember investigating if we could update the dataset quite some time ago, but it seemed like more modern data was defined differently, and it wasn't clear to me what updates to the code would be required.

**Comment by smarwin** [comment]
Hi Matthew,
thanks for answering so quickly. When checking the source code I also noticed that the parameter were quite old and I also noticed that I have less understanding of the matter than I need ;-)

I found this link where bvparams are available: https://www.iucr.org/resources/data/datasets/bond-valence-parameters

Maybe that helps!
